src/frameworks/index.ts

```diff
@@ -404,16 +406,23 @@ export async function prepareFrameworks(
       support,
     } = WebFrameworks[framework];
     console.log(`Detected a ${name} codebase. ${SupportLevelWarnings[support] || ""}\n`);
+    const hostingEmulatorInfo = emulators.find((e) => e.name === Emulators.HOSTING);
+
+    const frameworksBuild = getFrameworksBuildMode(options);
+
     // TODO allow for override
-    const isDevMode = context._name === "serve" || context._name === "emulators:start";
+    const isDevMode =
```

Comment:
can we do context._name !== "deploy" rather? also if !isDevMode let's ahead and set NODE_ENV=production

Reply:
Typescript complains with `Cannot assign to 'NODE_ENV' because it is a read-only property` if we try to assign something to `process.env.NODE_ENV`.